src/EditorFeatures/CSharpTest/MoveDeclarationNearReference/MoveDeclarationNearReferenceTests.cs

```diff
@@ -630,5 +630,593 @@ void M()
+
+        {
+            i = 0;
+            Console.Write(i);
+        }
+    }
+}",
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        if (false)
+        {
+        }
+
+        Console.WriteLine();
+
+        {
+            // leading trivia
+            int i = 0;
+            Console.Write(i);
+        }
+    }
+}",
+ignoreTrivia: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestComments07()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        if (false)
+        {
+        }
+
+        // leading trivia
+        int [||]i = 5;
+        Console.WriteLine();
+
+        // Existing trivia
+        i = 0;
+        Console.Write(i);
+    }
+}",
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        if (false)
+        {
+        }
+
+        Console.WriteLine();
+
+        // leading trivia
+        // Existing trivia
+        int i = 0;
+        Console.Write(i);
+    }
+}",
+ignoreTrivia: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestComments08()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        if (false)
+        {
+        }
+
+        // leading trivia
+        int [||]i = 5;
+        Console.WriteLine();
+
+        {
+            // Existing trivia
+            i = 0;
+            Console.Write(i);
+        }
+    }
+}",
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        if (false)
+        {
+        }
+
+        Console.WriteLine();
+
+        {
+            // leading trivia
+            // Existing trivia
+            int i = 0;
+            Console.Write(i);
+        }
+    }
+}",
+ignoreTrivia: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMergeComments01()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        // leading trivia
+        int [||]i = 5;
+        Console.WriteLine();
+
+        i = 0;
+        Console.Write(i);
+    }
+}",
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine();
+
+        // leading trivia
+        int i = 0;
+        Console.Write(i);
+    }
+}",
+ignoreTrivia: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMergeComments02()
+        {
+            await TestInRegularAndScriptAsync(
+@"class Program
+{
+    static void Main(string[] args)
+    {
+        // leading trivia
+        int [||]i = 5;
+        Console.WriteLine();
+
+        {
+            i = 0;
```

Comment:
No.  THere are later tests for that.  Merge-Comments simply means "testing comment behavior when we do a merge".  There are tests for comments on the decl, comments on the reference, and comments on both.